### PR TITLE
refactor: HepMC3 converter as namespace instead of struct

### PR DESCRIPTION
**Pull request by FabianKlimpel**
This PR changes the stateless HepMC3 converter structs to become namespaces instead.

### CODE REVIEW

**Comment by msmk0** [comment]
@FabianKlimpel Could you give a bit more details why this change is needed or is this purely a stylistic cleanup?

On a more general note: the `SimVertex`-based truth EDM in the examples is not used anymore. The only reason it has not been removed, was so as not to break the HepMC module. At the time of the EDM change, the HepMC module was not implemented as a reader/writer and not actively used. Since you are using the HepMC types directly without any conversion to/from the examples EDM, I would suggest that these `SimVertex`-based exporters be removed altogether or at least removed from the public interface of the HepMC module. What would be needed, would be the conversion of HepMC event(s) into a `SimParticles` container, such that HepMC files can be used as inputs to the simulation. This is not implemented by the current converters anyways.

File: Examples/Io/HepMC3/include/ActsExamples/Io/HepMC3/HepMC3Event.hpp [hunk 1]
```diff
@@ -21,156 +21,123 @@
 namespace ActsExamples {
 
 /// Helper struct to convert HepMC3 event to the internal format.
```
Comment (paulgessinger) [inline Examples/Io/HepMC3/include/ActsExamples/Io/HepMC3/HepMC3Event.hpp:23]:
```suggestion
/// Helper functions to convert HepMC3 event to the internal format.
```

File: Examples/Io/HepMC3/src/HepMC3Event.cpp [hunk 2]
```diff
@@ -11,6 +11,82 @@
...
+/// @note An Acts vertex does not store a barcode. Therefore the content of
+/// both vertices is compared. The position, time and number of incoming and
+/// outgoing particles will be compared. Since a second vertex could exist in
+/// the record with identical informations (although unlikely), this
+/// comparison could lead to false positive results. On the other hand, a
+/// numerical deviation of the parameters could lead to a false negative.
+/// @param actsVertex Acts vertex
+/// @param genVertex HepMC3::GenVertex
+/// @return boolean result if both vertices are identical
+bool compareVertices(const std::shared_ptr<ActsExamples::SimVertex>& actsVertex,
```
Comment (paulgessinger) [inline Examples/Io/HepMC3/src/HepMC3Event.cpp:69]: Not really important, but you could implement as a comparison operator, right?

File: Examples/Run/HepMC3/HepMC3Example.cpp [hunk 3]
```diff
@@ -56,37 +56,39 @@ int main(int argc, char** argv) {
   }
   std::cout << std::endl;
 
-  ActsExamples::HepMC3Event simEvent;
-
   std::cout << "Event data:" << std::endl;
   std::cout << "Units: ";
-  if (simEvent.momentumUnit(genevt) == Acts::UnitConstants::GeV)
+  if (ActsExamples::HepMC3Event::momentumUnit(genevt) ==
```
Comment (paulgessinger) [inline Examples/Run/HepMC3/HepMC3Example.cpp:61]: I guess you could do `using ActsExamples::HepMC3Event` to make this a bit shorter.

**Comment by paulgessinger** [comment]
@msmk0 I'm assuming this is based on the fact that the `struct`s before didn't have members, which calls into question if they need to be structs.

File: Examples/Io/HepMC3/src/HepMC3Event.cpp (hunk 2 above)
Comment (msmk0) [reply Examples/Io/HepMC3/src/HepMC3Event.cpp:69]: That would mean adding a comparison operator for types external to your library, right? I think having an explicit comparison function here is easier to understand.

**Review by msmk0 — APPROVED** [review]
@FabianKlimpel Looks ok.

**Comment by msmk0** [comment]
> @msmk0 I'm assuming this is based on the fact that the `struct`s before didn't have members, which calls into question if they need to be structs.

Yes, exactly.

File: Examples/Io/HepMC3/src/HepMC3Event.cpp (hunk 2 above)
Comment (paulgessinger) [reply Examples/Io/HepMC3/src/HepMC3Event.cpp:69]: ok

File: Examples/Io/HepMC3/include/ActsExamples/Io/HepMC3/HepMC3Event.hpp (hunk 1 above)
Comment (FabianKlimpel) [reply Examples/Io/HepMC3/include/ActsExamples/Io/HepMC3/HepMC3Event.hpp]: Done

File: Examples/Run/HepMC3/HepMC3Example.cpp (hunk 3 above)
Comment (FabianKlimpel) [reply Examples/Run/HepMC3/HepMC3Example.cpp]: Done

**Review by msmk0 — APPROVED** [review]

**Pull request merged**